### PR TITLE
D3 Overlay shouldComponentUpdate methods

### PR DESCRIPTION
**Pull request by volfied**
- For superselect, policy and sampler overlays
- Correct the guides overlay existing SCU
- Change Sampler behavior so it highlgihts/samples leaf layers
- Re-add file watcher that just prints out file changes

Addresses #3491, and some sampler behavior issues.

### CODE REVIEW

File: src/js/jsx/tools/SuperselectOverlay.jsx [hunk 1]
```diff
@@ -126,6 +127,21 @@ define(function (require, exports, module) {
...
 
+        shouldComponentUpdate: function (nextProps, nextState) {
+            var lastLayers = this.state.document.layers.selected,
+                nextLayers = nextState.document.layers.selected,
+                lastLayerProps = collection.pluckAll(lastLayers, ["id", "bounds"]),
+                nextLayerProps = collection.pluckAll(nextLayers, ["id", "bounds"]),
+                layersChanged = !Immutable.is(lastLayerProps, nextLayerProps);
+
+            return layersChanged ||
+                this.state.marqueeEnabled !== nextState.marqueeEnabled ||
```
Comment (iwehrman) [inline src/js/jsx/tools/SuperselectOverlay.jsx]: Would it make more sense to do these faster checks first before falling back to the heavier check based on pluck?

**Comment by iwehrman** [comment]
Looks fine otherwise. One Q for @volfied.

**Comment by volfied** [comment]
@iwehrman right back at ya.

**Comment by iwehrman** [comment]
ok!

**Pull request merged**